src/server/scripts/EasternKingdoms/Stratholme/instance_stratholme.cpp

```diff
@@ -82,6 +86,33 @@ class instance_stratholme : public InstanceMapScript
             GuidSet abomnationGUID;
             EventMap events;
 
+            void OnUnitDeath(Unit* who) override
+            {
+                switch (who->GetEntry())
+                {
+                    case NPC_CRIMSON_GUARDSMAN:
+                    case NPC_CRIMSON_CONJUROR:
+                    case NPC_CRIMSON_INITATE:
+                    case NPC_CRIMSON_GALLANT:
+                    {
+                        if (!timmySpawned)
+                        {
+                            Position pos = who->ToCreature()->GetHomePosition();
+                            // check if they're in front of the entrance
+                            if (pos.GetPositionX() > scarletEntrancePositionX && pos.GetPositionY() < scarletEntrancePositionY)
```

Comment:
Position::IsWithinDist ?

Reply:
Hm, could work but they're spread out in a way that if we set a constant distance then monsters that are inside the castle might count as well and from what i've read its only those outside the building

Reply:
Boundary might come handy then
You can create a boundary in the instance and check if the creature dies within it by calling ```IsWithinBoundaryArea```

Reply:
A little bit of overkill for such a simple thing in my opinion but i'll look into it :P

Reply:
not really, predefined boundaries like rectangles or circles are already coded, you would just create an instance of it